.buildkite/integration.pipeline.yml

```diff
@@ -108,13 +113,16 @@ steps:
           PLATFORMS: "linux/amd64"
         command: |
           .buildkite/scripts/steps/integration-package.sh
+          .buildkite/scripts/steps/integration-cloud-image-push.sh
```

Comment:
Iirc when having two separate commands like this if the first fails and the next succeeds the step won't fail.

I think we need either `&&` or `set -euo pipefail` as the first statement (the latter will fail the step on any error inside any script not just based on the exit code though)

Same comment for the rest

Reply:
thanks for the point @dliappis, I had the impression that this wasn't required https://github.com/elastic/elastic-agent/pull/8337#discussion_r2195300630. But if it is, I need to do a follow-up PR and start from main and then bring over 9.1 and 8.19. wdyt?

Reply:
> had the impression that this wasn't required [#8337 (comment)](https://github.com/elastic/elastic-agent/pull/8337#discussion_r2195300630). But if it is, I need to do a follow-up PR and start from main and then bring over 9.1 and 8.19. wdyt?

I wouldn't do `set -euo pipefail`, but **if memory serves right** we may need to `.buildkite/scripts/steps/integration-package.sh && .buildkite/scripts/steps/integration-cloud-image-push.sh` to ensure if either fails, the step fails too. I'll try a test later today with a PR somewhere to confirm if we need to. For now I am good with things as they are.

Reply:
no worries @dliappis, I see what you mean and let's go with better safe than sorry so I will open a follow up PR to go with && approach